### PR TITLE
Module Tags, refactoring code repository

### PR DESCRIPTION
**Pull request by DimaKacer**

### CODE REVIEW

File: angular-spa-ui/js/package.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
```
Comment (tarhunakau) [inline angular-spa-ui/js/package.json]: Why you created this file here? It must be in "angular-spa-ui/package.json".

File: angular-spa-ui/js/search-app/tags-module/index.js [hunk 2]
```diff
@@ -2,16 +2,35 @@
...
+                controller: tagsCtrl
+            });
     };
 
+    function tagIteme() {
+        return {
+            restrict: 'E',
+            templateUrl: 'tagItemTemplate.html',
+            link: function(scope, element, attrs) {
+                //console.log(scope)
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/tags-module/index.js]: Commented code in pull request allowed in next situations:
1. That is a feature for future
2. That is a feature that should be disabled at the moment for some time
3. That is a quick fix for immediately testing

Please remove commented code

File: angular-spa-ui/js/search-app/tags-module/index.js [hunk 3]
```diff
@@ -2,16 +2,35 @@
...
+                element.bind('click', function(el) {
+                    console.log(el.target)
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/tags-module/index.js]: Do you really need console.log()?

File: angular-spa-ui/js/search-app/tags-module/tags-cfg.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+var stateProvider = function($stateProvider) {
+    $stateProvider
+        .state('tags', {
+            url: '/tags',
+            templateUrl: '/js/search-app/tags-module/tags.jade',
+            controller: tagsCtrl
+        });
+}
+
+module.exports = stateProvider;
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/tags-module/tags-cfg.js]: We decided that module should looks next way:

module.exports = function ([$arg1, ..., $argn]) {
    // Your code here
}

File: angular-spa-ui/js/search-app/tags-module/tags-ctrl.js [hunk 5]
```diff
@@ -1 +1,81 @@
 "use strict";
+
+//var tagsCtrl = {};
+
+module.exports = function($scope, $http) {
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/tags-module/tags-ctrl.js:3]: When this module initializes it send 7 _GET requests to the server. Why you did it?

File: angular-spa-ui/js/search-app/tags-module/tags-dir.js [hunk 6]
```diff
@@ -1 +1,7 @@
 "use strict";
+
+function tagItem($scope) {
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/tags-module/tags-dir.js]: module.exports = function () {}

File: angular-spa-ui/js/search-app/tags-module/templates/tag-item.html [hunk 7]
```diff
@@ -0,0 +1,3 @@
+<script id="tagItemTemplate.html" type="text/ng-template">
+    <span class="tag" tid="{{tag._id}}" style="color: {{tag.textColor}}; background-color:{{tag.backgroundColor}};" >{{tag.name}}</span>
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/tags-module/templates/tag-item.html]: Comment for Friday:
Variable in '{{' '}}' brackets should be separated with space:
{{ tag.backgroundColor }}

**Comment by tarhunakau** [comment]
Must be refactored

**Comment by DimaKacer** [comment]
All fixed

File: angular-spa-ui/js/search-app/tags-module/tags-ctrl.js [hunk 8]
```diff
@@ -1 +1,15 @@
 "use strict";
+
+module.exports = function($scope, $http) {
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/tags-module/tags-ctrl.js:3]: As I remember, we decided to move _GET request to service. You can read this sample: http://stackoverflow.com/questions/17646034/what-is-the-best-practice-for-making-an-ajax-call-in-angular-js

File: angular-spa-ui/js/search-app/tags-module/tags-dir.js [hunk 9]
```diff
@@ -1 +1,20 @@
 "use strict";
+
+module.exports = function(moduleTags) {
+
+    moduleTags
+        .directive('tagIteme', tagIteme);
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/tags-module/tags-dir.js:6]: tagIteme? Looks like misprint. Check it.

**Comment by tarhunakau** [comment]
To be refactored

**Pull request closed without merging**